openmc/stats/multivariate.py

```diff
@@ -0,0 +1,543 @@
+    reference_uvw : Iterable of float
+        Direction from which polar angle is measured. Defaults to the positive
+        x-direction.
+
+    """
+
+
+    def __init__(self, reference_uvw=[1., 0., 0.]):
+        super().__init__(reference_uvw)
+
+    def to_xml_element(self):
+        """Return XML representation of the monodirectional distribution
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing monodirectional distribution data
+
+        """
+        element = ET.Element('angle')
+        element.set("type", "monodirectional")
+        if self.reference_uvw is not None:
+            element.set("reference_uvw", ' '.join(map(str, self.reference_uvw)))
+        return element
+
+    @classmethod
+    def from_xml_element(cls, elem):
+        """Generate monodirectional distribution from an XML element
+
+        Parameters
+        ----------
+        elem : xml.etree.ElementTree.Element
+            XML element
+
+        Returns
+        -------
+        openmc.stats.Monodirectional
+            Monodirectional distribution generated from XML element
+
+        """
+        monodirectional = cls()
+        params = get_text(elem, 'parameters')
+        if params is not None:
+            monodirectional.reference_uvw = [float(x) for x in params.split()]
+        return monodirectional
+
+
+class Spatial(metaclass=ABCMeta):
+    """Distribution of locations in three-dimensional Euclidean space.
+
+    Classes derived from this abstract class can be used for spatial
+    distributions of source sites.
+
+    """
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def to_xml_element(self):
+        return ''
+
+    @classmethod
+    @abstractmethod
+    def from_xml_element(cls, elem):
+        distribution = get_text(elem, 'type')
+        if distribution == 'cartesian':
+            return CartesianIndependent.from_xml_element(elem)
+        elif distribution == 'box' or distribution == 'fission':
+            return Box.from_xml_element(elem)
+        elif distribution == 'point':
+            return Point.from_xml_element(elem)
+
+
+class CartesianIndependent(Spatial):
+    """Spatial distribution with independent x, y, and z distributions.
+
+    This distribution allows one to specify a coordinates whose x-, y-, and z-
+    components are sampled independently from one another.
+
+    Parameters
+    ----------
+    x : openmc.stats.Univariate
+        Distribution of x-coordinates
+    y : openmc.stats.Univariate
+        Distribution of y-coordinates
+    z : openmc.stats.Univariate
+        Distribution of z-coordinates
+
+    Attributes
+    ----------
+    x : openmc.stats.Univariate
+        Distribution of x-coordinates
+    y : openmc.stats.Univariate
+        Distribution of y-coordinates
+    z : openmc.stats.Univariate
+        Distribution of z-coordinates
+
+    """
+
+
+    def __init__(self, x, y, z):
+        super().__init__()
```

Comment:
Spatial's \_\_init\_\_ function doesn't do anything, so this line doesn't add anything to the method. Can be removed and ditto throughout.